### PR TITLE
[Enhancement](alter) support add partitions in batch

### LINKED ISSUE #32524: [Enhancement] support add partitions in batch
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

ALTER TABLE db.table ADD PARTITIONS FROM() TO()  INTERVAL num [UNIT]
e.g.
ALTER TABLE db.table ADD PARTITIONS FROM(1) TO(10)  INTERVAL 1
e.g.
ALTER TABLE db.table ADD PARTITIONS FROM("2024-01") TO("2024-12-31")  INTERVAL 1 MONTH

### Solution

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by felixwluo**
## Proposed changes

Issue Number: close #32524

### CODE REVIEW

**Comment by felixwluo** [comment]
run buildall

**Review by cambyzju — DISMISSED** [review]

**Review by lide-reed — DISMISSED** [review]
LGTM

**Review by xy720 — DISMISSED** [review]

**Comment by xy720** [comment]
run buildall

**Comment by felixwluo** [comment]
run buildall

File: regression-test/suites/alter_p0/test_alter_add_multi_partition.groovy [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        )
+        DISTRIBUTED BY HASH(`user_id`) BUCKETS 3
+        PROPERTIES
+        (
+            "replication_num" = "1"
+        );
+    """
+    List<List<Object>> result1  = sql "show partitions from add_multi_partition;"
+    assertEquals(result1.size(), 1)
+    sql "ALTER TABLE add_multi_partition ADD PARTITIONS FROM ('2019-01-01') TO ('2025-01-01') INTERVAL 1 YEAR;"
```
Comment (yiguolei) [inline regression-test/suites/alter_p0/test_alter_add_multi_partition.groovy:44]: the test is too simple.
Should test more complicated case.
For example, partition by day, then the partition interval is 5 day, to test if we could deal with Feb 29 or Aug 31.
And also test, if could support 3min

**Comment by felixwluo** [comment]
run buildall

**Comment by felixwluo** [comment]
run buildall

File: regression-test/suites/alter_p0/test_alter_add_multi_partition.groovy (hunk 1 above)
Comment (felixwluo) [reply regression-test/suites/alter_p0/test_alter_add_multi_partition.groovy]: added 0229 partition. No problem

File: fe/fe-core/src/main/java/org/apache/doris/alter/Alter.java [hunk 2]
```diff
@@ -506,6 +508,8 @@ public void processAlterTable(AlterTableStmt stmt) throws UserException {
             } else if (alterClause instanceof ModifyTablePropertiesClause) {
                 Map<String, String> properties = alterClause.getProperties();
                 ((SchemaChangeHandler) schemaChangeHandler).updateTableProperties(db, tableName, properties);
+            } else if (alterClause instanceof AlterMultiPartitionClause) {
+                Env.getCurrentEnv().addMultiPartitions(db, tableName, (AlterMultiPartitionClause) alterClause);
```
Comment (yujun777) [inline fe/fe-core/src/main/java/org/apache/doris/alter/Alter.java:512]: need check allow alter,   see line. 484 ~ 493

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java [hunk 3]
```diff
@@ -1801,6 +1803,24 @@ public void addPartition(Database db, String tableName, AddPartitionClause addPa
...
+                    multiPartitionClause.getProperties());
+            singlePartitionDescs = multiPartitionDesc.getSinglePartitionDescList();
+        } catch (AnalysisException e) {
+            throw new DdlException("Failed to analyze multi partition clause: " + e.getMessage());
+        }
+
+        for (SinglePartitionDesc singlePartitionDesc : singlePartitionDescs) {
+            AddPartitionClause addPartitionClause = new AddPartitionClause(singlePartitionDesc, null,
+                    multiPartitionClause.getProperties(), false);
+            addPartition(db, tableName, addPartitionClause);
```
Comment (yujun777) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java:1820]: addPartition's args had changed,   pls check the lastest code.

**Review by yujun777 — CHANGES_REQUESTED** [review]
need update

**Review by yujun777 — APPROVED** [review]
LGTM

**Review by deardeng — APPROVED** [review]
LGTM

**Review by cambyzju — APPROVED** [review]

**Review by yiguolei — APPROVED** [review]

**Comment by dataroaring** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**